agent/agent-profiler/docs/README.md

```diff
@@ -1,101 +1,120 @@
+
+1. Inside the `applicationinsights.json` configuration of your process enable the profiler by
+   setting the `preview.profiler.enabled` setting:
+   ```json
+      {
+         "connectionString" : "...",
+         "preview" : {
+            "profiler" : {
+               "enabled" : true
+            }
+         }
+      }
+   ```
+   Alternatively, set the `APPLICATIONINSIGHTS_PROFILER_ENABLED` environment variable to true.
+2. Execute your process with the updated configuration.
+3. Configure the resource thresholds that will cause a profile to be collected:
+    1. Browse to the Performance -> Profiler section of the associated Application Insights instance.
+    2. Select "Triggers"
+    3. Configure the required CPU and Memory thresholds. And Apply.
+    4. Note, currently the Java profiler does not
+   support the "Sampling" trigger, configuring this will have no effect.
+
+Once this has been completed, the agent will monitor the resource usage of your processes and
+trigger a profile when required. Once a profile has been triggered and completed, it will be
+viewable from the
+Application Insights instance within the Performance -> Profiler section. From that screen the
+profile can be downloaded, once download the JFR recording file can be opened and analysed within a
+tool of the users choosing.
 
 ## Configuration
 
-Configuration of the profiler triggering settings, such as thresholds and profiling periods can be performed within the
-ApplicationInsights UI under the Performance, Profiler, Triggers UI.
+Configuration of the profiler triggering settings, such as thresholds and profiling periods can be
+performed within the ApplicationInsights UI under the Performance, Profiler, Triggers UI as
+described in [Installation](#Installation).
+
+Additionally, a number of parameters can be configured within environment variables and the
+`applicationinsights.json` configuration file.
+
+### Environment variables
 
 - APPLICATIONINSIGHTS_PROFILER_ENABLED: boolean (default false)
-    - Can be set as an environment variable to enable the profiler.
-    - While profiling is in beta it must be actively enabled.
-- APPLICATIONINSIGHTS_PREVIEW_METRIC_INTERVAL_SECONDS: int (default 60)
-    - Can be set as an environment variabl.
-    - Period in seconds that performance counters are sampled. This directly impacts the frequency that
-      Alerting thresholds are evaluated. Alerting thresholds by default apply an average over the last 2 min,
-      at the default of 60 seconds this will be 2 samples.
+  - Enables/Disables the profiling feature.
 
-Profiler may also be enabled via adding a config section to applicationinsights.json:
+### Configuration file
 
-```
+```json
 {
   "preview": {
     "profiler": {
-      "configPollPeriodSeconds": 60,
-      "enabled": true
+      "enabled": true,
+      "configPollPeriodSeconds": 50,
+      "cpuTriggeredSettings": "profile_without_env_data",
+      "memoryTriggeredSettings": "profile_without_env_data"
```

Comment:
I noticed this is uppercase in other places, tho lowercase (with dashes) is more consistent with other property "enum" values in the config would be nice to change for consistency, e.g. https://github.com/microsoft/ApplicationInsights-Java/blob/020a55539cae09c09736a62ac671345c8b962102/agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java#L393-L400)